prolif/protein_helper.py

```diff
@@ -0,0 +1,560 @@
+        actual protonated states of residues.
+        For example, neutral and protonated arginine are both assigned to ARG,
+        while neutral and deprotonated arginine in GROMOS force field
+        are assigned to ARGN and ARG, respectively.
+
+
+        Parameters
+        ----------
+        resname : str
+            The residue name to convert.
+        forcefield_name : str
+            The name of the forcefield to use for conversion. Default is "unknown".
+
+        Returns
+        -------
+        str
+            The standard residue name.
+        """
+
+        if forcefield_name == "unknown":
+            warnings.warn(
+                "Could not guess the forcefield based on the residue names. "
+                "CYS is assigned to neutral CYS (charge = 0).",
+                stacklevel=2,
+            )
+
+        elif forcefield_name == "gromos" and resname == "CYS":
+            return "CYX"
+
+        return resnames_map.get(resname, resname)
+
+    @staticmethod
+    def check_resnames(
+        resnames_to_check: set[str], templates: list[dict] | None = None
+    ) -> bool:
+        """Check if the residue names are standard or in templates and
+        raise a warning if not.
+
+        Parameters
+        ----------
+        resnames_to_check : set[str]
+            Set of residue names to check.
+
+        Raises
+        ------
+        UserWarning
+            If any residue name is not standard or not in the templates.
+
+        Returns
+        -------
+        bool
+            True if all residue names are standard or in the templates,
+            False otherwise.
+        """
+
+        if templates is None:
+            templates = [STANDARD_AA]
+
+        all_available_resnames = set()
+        for t in templates:
+            all_available_resnames.update(t.keys())
+
+        non_standard_resnames = resnames_to_check - all_available_resnames
+        if non_standard_resnames:
+            warnings.warn(
+                "Non-standard residue (or ligand/solvent) names found "
+                f"not in templates: {non_standard_resnames}",
+                stacklevel=2,
+            )
+            return False
+        return True
+
+    @staticmethod
+    def n_residue_heavy_atoms(residue: Residue) -> int:
+        """Count the number of heavy atoms in a residue.
+
+        Parameters
+        ----------
+        residue : Residue
+            The residue to count the heavy atoms in.
+
+        Returns
+        -------
+        int
+            The number of heavy atoms in the residue.
+        """
+        terminal_oxygen = {
+            key
+            for key, value in ATOMNAME_ALIASES["Protein"].items()
+            if value.strip() == "OXT"
+        }
+        terminal_oxygen.add("OXT")
+        return len(
+            [
+                atom
+                for atom in residue.GetAtoms()
+                if atom.GetSymbol().strip() != "H"
```

Comment:
no need for `strip` on `GetSymbol`, but I'd rather see `GetAtomicNum` instead
```suggestion
                if atom.GetAtomicNum() != 1
```